pkg/cloud/awsprovider.go

```diff
@@ -161,20 +158,42 @@ type AWS struct {
 	*CustomProvider
 }
 
+// AWSAccessKey holds AWS credentials and fulfils the awsV2.CredentialsProvider interface
 type AWSAccessKey struct {
 	AccessKeyID     string `json:"aws_access_key_id"`
 	SecretAccessKey string `json:"aws_secret_access_key"`
 }
 
 // Retrieve returns a set of awsV2 credentials using the AWSAccessKey's key and secret.
-// This fullfils the awsV2.CredentialsProvider interface contract.
-func (accessKey AWSAccessKey) Retrieve(ctx context.Context) (awsV2.Credentials, error) {
-	return awsV2.Credentials{
+// This fulfils the awsV2.CredentialsProvider interface contract.
+func (accessKey AWSAccessKey) Retrieve(ctx context.Context) (awsSDK.Credentials, error) {
+	return awsSDK.Credentials{
 		AccessKeyID:     accessKey.AccessKeyID,
 		SecretAccessKey: accessKey.SecretAccessKey,
 	}, nil
 }
 
+// CreateConfig creates an AWS SDK V2 Config for the credentials that it contains for the provided region
+func (accessKey AWSAccessKey) CreateConfig(region string) (awsSDK.Config, error) {
+	var cfg awsSDK.Config
+	var err error
+	// If accessKey values have not been provided, attempt to load cfg from service key annotations
+	if accessKey.AccessKeyID == "" && accessKey.SecretAccessKey == "" {
+		cfg, err = config.LoadDefaultConfig(context.TODO(), config.WithRegion(region))
+		if err != nil {
+			return cfg, fmt.Errorf("failed to initialize AWS SDK config for region from annotation %s: %s", region, err)
+		}
+	} else {
+		// The AWS SDK v2 requires an object fulfilling the CredentialsProvider interface, which cloud.AWSAccessKey does
+		cfg, err = config.LoadDefaultConfig(context.TODO(), config.WithCredentialsProvider(accessKey), config.WithRegion(region))
+		if err != nil {
+			return cfg, fmt.Errorf("failed to initialize AWS SDK config for region %s: %s", region, err)
+		}
+	}
+
+	return cfg, nil
+}
+
```

Comment:
Base config function which should be used by all non athena SDK v2 methods takes into account possibility of annotation configuration

Reply:
So, this config will always use annotation configuration values? Do you know the order in which it evaluates auth methods?

Reply:
So this does prioritize set ENV variables, however the methods which retrieve the AccessKeyID and SecretAccessKey set those env variables so they should not be set. The list of priority is here. https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/#specifying-credentials. This does leave the door open to other config methods if the use should so choose, but I think it is fair to say that if you want to use the service account annotation configuration you should not have erroneous other configuration methods floating around in you container/deployment.